src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1567 @@
+            labels_unique = set(labels)
+            self.GraphicLayerSequence = graphic_layers
+
+        if graphic_annotations is not None:
+            for ann in graphic_annotations:
+                if not isinstance(ann, GraphicAnnotation):
+                    raise TypeError(
+                        'Items of "graphic_annotations" must be of type '
+                        'highdicom.pr.GraphicAnnotation.'
+                    )
+                if ann.GraphicLayer not in labels_unique:
+                    raise ValueError(
+                        f'Graphic layer with name "{ann.GraphicLayer}" is '
+                        'referenced in "graphic_annotations" but not '
+                        'included "graphic_layers".'
+                    )
+                for item in ann.ReferencedImageSequence:
+                    uids = (
+                        item.ReferencedSOPClassUID,
+                        item.ReferencedSOPInstanceUID
+                    )
+                    if uids not in ref_uids:
+                        raise ValueError(
+                            f'Instance with SOP Instance UID {uids[1]} and '
+                            f'SOP Class UID {uids[0]} is referenced in '
+                            'items of "graphic_layers", but not included '
+                            'in "referenced_images".'
+                        )
+                for obj in getattr(ann, 'GraphicObjectSequence', []):
+                    grp_id = obj.graphic_group_id
+                    if grp_id is not None:
+                        if grp_id not in described_groups_ids:
+                            raise ValueError(
+                                'Found graphic object with graphic group '
+                                f'ID "{grp_id}", but no such group is '
+                                'described in the "graphic_groups" '
+                                'argument.'
+                            )
+                for obj in getattr(ann, 'TextObjectSequence', []):
+                    grp_id = obj.graphic_group_id
+                    if grp_id is not None:
+                        if grp_id not in described_groups_ids:
+                            raise ValueError(
+                                'Found text object with graphic group ID '
+                                f'"{grp_id}", but no such group is '
+                                'described in the "graphic_groups" '
+                                'argument.'
+                            )
+            self.GraphicAnnotationSequence = graphic_annotations
+
+        # Displayed Area Selection Sequence
+        # This implements the simplest case - all images are unchanged
+        # May want to generalize this later
+        display_area_item = Dataset()
+        display_area_item.ReferencedImageSequence = ref_im_seq
+        display_area_item.PixelOriginInterpretation = 'FRAME'
+        display_area_item.DisplayedAreaTopLeftHandCorner = [1, 1]
```

Comment:
> FRAME or VOLUME are units for image coordinates (a.k.a. 2D spatial coordinates) and these values should be specified at sub-pixel resolution in range [0, Total Pixel Matrix Columns] and [0, Total Pixel Matrix Rows]. In the image coordinate system, the TLHC pixel is located at (0.5, 0.5) and the BRHC pixel at (Total Pixel Matrix Columns - 0.5, Total Pixel Matrix Rows - 0.5).

Not so sure about this. The meaning of PixelOriginInterpretation within the module does mention coordinates or sub-pixel resolution see [here]https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.10.4.html#table_C.10-4) (though this may be different elsewhere).

> The Displayed Area Top Left Hand Corner and Displayed Area Bottom Right Hand Corner specify pixel indices. However, these indices are zero-based and should thus be (0, 0) rather than (1, 1).

Yeah this is different from elsewhere, very confusing. However it is pretty clear in the standard that (1,1) is what should go here. See [here](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.10.4.html#figure_C.10.4-1)

Reply:
This is so screwed up! [Figure C.10.4-1](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.10.4.html#figure_C.10.4-1) as well as the descriptions of Displayed Area Top Left Hand Corner and Displayed Area Bottom Right Hand Corner are plain wrong in my opinion. The index of the TLHC pixel should be (0, 0) and not (1, 1).

@dclunie what shall we do about that? I don't think we can change the standard to use the correct values. However, we should at least fix the description and clarify that these values are neither pixel indices (which are zero-based integer values) nor image coordinates (which are zero-based floating point values). This means that neither of the equations specified in the standard can be used to convert these coordinates. We would need yet another equation that first translates the points by 1 pixel to convert the one-based pixel indices into zero-based indices or vice versa.

Note that the description of Column Position In Total Image Pixel Matrix and Row Position In Total Image Pixel Matrix attributes is similarly wrong [C.8.12.6.1 Plane Position (Slide) Macro](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.8.12.6.html#sect_C.8.12.6.1).

Reply:
@CPBridge I don't think there is anything we could/should do at this point in highdicom. Do you agree?

Reply:
After further discussion, we concluded that image pixel positions in the standard are generally one-based indices. Therefore, the  Displayed Area Top Left Hand Corner and Displayed Area Bottom Right Hand Corner are not wrong. However, the values of these attributes will need to be converted into zero-based integer index values when using them with Equation C.7.6.2.1-1.

Reply:
I will resolve this comment. We can discuss separately whether we want to further elaborate on these details in the documentation (within the `pr` package or the `spatial` module)